pkg/volume/volume.go

```diff
@@ -239,7 +241,7 @@ type DeviceMounter interface {
 	// MountDevice mounts the disk to a global path which
 	// individual pods can then bind mount
 	// Note that devicePath can be empty if the volume plugin does not implement any of Attach and WaitForAttach methods.
-	MountDevice(spec *Spec, devicePath string, deviceMountPath string) error
+	MountDevice(spec *Spec, devicePath string, deviceMountPath string) (volumetypes.OperationStatus, error)
```

Comment:
As far as I can tell the purpose of `volumetypes.OperationStatus` is to differentiate between 3 possible states: 1) success/no error (state certain), 2) error failed operation (state certain), and 3) error failed operation (uncertain state)?

There are two pieces of information here: 1) `error`/`no error`,  and 2) for `no error`, there are two possible sub-states: `certain`/`uncertain`.

Returning `error` AND an explicit `volumetypes.OperationStatus` seems wrong since it allows for the possibility of the two parameters to contradict each other.

I think typed errors would be cleaner here. But overall this is a nit.

Reply:
fixed